packages/firestore/src/local/local_documents_view.ts

```diff
@@ -74,6 +77,25 @@ export class LocalDocumentsView {
     });
   }
 
+  // Returns the view of the given `docs` as they would appear after applying
+  // all mutations in the given `batches`.
+  private applyLocalMutationsToDocuments(
+    transaction: PersistenceTransaction,
+    docs: NullableMaybeDocumentMap,
+    batches: MutationBatch[]
+  ): PersistencePromise<NullableMaybeDocumentMap> {
+    let results = nullableMaybeDocumentMap();
+    return new PersistencePromise<NullableMaybeDocumentMap>(resolve => {
```

Comment:
@var-const In general, JavaScript is completely single-threaded (there's no way to block other than by spinning the CPU) and so wrapping a computation in a promise doesn't really help (in particular it won't enable any parallelism).

And PersistencePromise is extra weird because it's a specially-designed Promise-like construct that tries to be as synchronous as possible because IndexedDb has weird semantics where in the completion for one operation you *must* synchronously start the next operation or else your transaction will auto-close.  So even using PersistencePromise, this code is actually 100% synchronous.  So you can go ahead and just yank PersistencePromise out.